When finding the full set of relations between variants, the graph will quickly become very dense since it will contain $n^2$ edges for $n$ nodes. 
A substantial reduction in the amount of edges can be achieved by removing the disjoint relations. 
An absent edge between two nodes is implicitly understood as representing a disjoint relation. 
Similarly, one arbitrary direction of the containment relation can be removed since the other direction follows implicitly. 
In this thesis the \textit{is contained} direction is kept.
Based on the definitions of the relations, many more edges can be pruned from the graph without loss of information.
In Figure \ref{fig:Pruning} the different instances in which relations can be pruned are explained.
%
% TODO generate nicer images
% TODO fix underfull warnings
% TODO make cyp2d6 a command?
%
\begin{figure}
    \newcommand{\subfigurelength}{0.35\textwidth} % TODO move this command?
    \centering
    \begin{subfigure}{\subfigurelength}
        \centering
        \includesvg[width=\textwidth]{relations/pruning/reflexive.svg}
        \caption{Reflexivity. 
            The equivalence relation is reflexive. 
            This means that every allele and variant is equivalent to itself. 
            This is understood implicitly and thus doesn't have to be represented in the graph.
        }
        \label{sfig:Reflexivity}
    \end{subfigure}
    \begin{subfigure}{\subfigurelength}
        \centering
        \includesvg[width=\textwidth]{relations/pruning/symmetric.svg}
        \caption{Symmetry. 
            The equivalence, overlap and disjoint relation are symmetric. 
            Only one edge between a pair of nodes is needed to represent these relations.
        }
        \label{sfig:Symmetry}
    \end{subfigure}
    \begin{subfigure}{\subfigurelength}
        \centering
        \includesvg[width=\textwidth]{relations/pruning/transitive.svg}
        \caption{Transitivity. 
            Both equivalence and containment relation are transitive. 
            A transitive reduction can be applied to the graph to remove redundant edges. 
            In this example CYP2D6*39 is contained in CYP2D6*10 which is contained in CYP2D6*147. 
            The transitive reduction removes the edge between CYP2D6*39 and CYP2D6*147.
        }
        \label{sfig:Transitivity}
    \end{subfigure}
    \begin{subfigure}{\subfigurelength}
        % TODO name this reduction
        \centering
        % \includesvg[width=\textwidth]{relations/pruning/most-specific.svg}
        \caption{Equivalence.
            % TODO
        }
        \label{sfig:Equivalence-Reduction}
    \end{subfigure}
    \begin{subfigure}{\subfigurelength}
        % TODO rename this reduction?
        \centering
        \includesvg[width=\textwidth]{relations/pruning/common-ancestor.svg}
        \caption{Common Ancestor. 
            Given a graph where both transitive reduction (see \ref{sfig:Transitivity}) and equivalence reduction (see \ref{sfig:Equivalence-Reduction}) have been applied, any pair of nodes that contain the same node must overlap. 
            In this example both CYP2D6*109 and CYP2D6*115 contain CYP2D6*9. 
            Therefore they must overlap. 
            An important footnote is that CYP2D6*9 does not characterize the overlap. % TODO explain further?
        }
        \label{sfig:Common-Ancestor}
    \end{subfigure}
    \begin{subfigure}{\subfigurelength}
        \centering
        \includesvg[width=\textwidth]{relations/pruning/most-specific.svg}
        \caption{Most Specific.
            % TODO
        }
        \label{sfig:Most-Specific}
    \end{subfigure}
    \caption{Different instances in which relations can be pruned with specific examples for the gene CYP2D6.}
    \label{fig:Pruning}
\end{figure}
